src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,51 @@
+package mate.academy.service.impl;
+
+import java.util.ArrayList;
+import mate.academy.dao.ShoppingCartDao;
+import mate.academy.dao.TicketDao;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.MovieSession;
+import mate.academy.model.ShoppingCart;
+import mate.academy.model.Ticket;
+import mate.academy.model.User;
+import mate.academy.service.ShoppingCartService;
+
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+    @Inject
+    private TicketDao ticketDao;
+
+    @Override
+    public void addSession(MovieSession movieSession, User user) {
+        Ticket ticket = new Ticket();
+        ticket.setUser(user);
+        ticket.setMovieSession(movieSession);
+        ticketDao.add(ticket);
+
+        ShoppingCart shoppingCart = shoppingCartDao.getByUser(user).get();
+        shoppingCart.getTickets().add(ticket);
+        shoppingCartDao.update(shoppingCart);
+    }
+
+    @Override
+    public ShoppingCart getByUser(User user) {
+        return shoppingCartDao.getByUser(user).orElseThrow(
+                () -> new RuntimeException("Can't find user " + user));
```

Comment:
```suggestion
                () -> new RuntimeException("Can't find shopping cart by user: " + user));
```
here need to search for shoppingCart, not user